wcfsetup/install/files/lib/data/style/Style.class.php

```diff
@@ -232,14 +245,20 @@ public function getCoverPhoto() {
 	 * @since 5.2
 	 */
 	public function getCoverPhotoLocation() {
-		return WCF_DIR . 'images/coverPhotos/' . $this->getCoverPhoto();
+		if ($this->coverPhotoExtension) {
+			return $this->getAssetPath().'coverPhoto.'.$this->coverPhotoExtension;
+		}
+		return WCF_DIR . 'images/coverPhotos/default.jpg';
 	}
 	
 	/**
 	 * @return string
 	 * @since 5.2
 	 */
 	public function getCoverPhotoUrl() {
+		if ($this->coverPhotoExtension) {
+			return FileUtil::getRelativePath(WCF_DIR, $this->getAssetPath()).'coverPhoto.'.$this->coverPhotoExtension;
```

Comment:
This is incorrect, `WCF::getPath()` returns an absolute URL to the specified app (implicit: the core), but your code only yields a path relative to the Core's directory.

Reply:
You are entirely correct here, of course.